backend/regalloc/regalloc_irc.ml

```diff
@@ -62,7 +62,7 @@ let build : State.t -> Cfg_with_infos.t -> unit =
       if is_move_instruction instr
          && (not (Reg.is_stack instr.arg.(0)))
          && (not (Reg.is_stack instr.res.(0)))
-         && same_reg_class instr.arg.(0) instr.res.(0)
+         && Proc.types_are_compatible instr.arg.(0) instr.res.(0)
```

Comment:
(I think this is slightly more constrained,
which is fine - probably even better.)